pkg/util/helper/predicate.go

```diff
@@ -17,74 +18,39 @@ import (
 func NewExecutionPredicate(mgr controllerruntime.Manager) predicate.Funcs {
-	return predicate.Funcs{
-		CreateFunc: func(createEvent event.CreateEvent) bool {
-			obj := createEvent.Object.(*workv1alpha1.Work)
+	predFunc := func(eventType string, object client.Object) bool {
+		obj := object.(*workv1alpha1.Work)
 
-			// Ignore the object that has been suppressed.
-			if util.GetLabelValue(obj.Labels, util.PropagationInstruction) == util.PropagationInstructionSuppressed {
-				klog.V(5).Infof("Ignored Work(%s/%s) create event as propagation instruction is suppressed.", obj.Namespace, obj.Name)
-				return false
-			}
+		// Ignore the object that has been suppressed.
+		if util.GetLabelValue(obj.Labels, util.PropagationInstruction) == util.PropagationInstructionSuppressed {
+			klog.V(5).Infof("Ignored Work(%s/%s) %s event as propagation instruction is suppressed.", obj.Namespace, obj.Name, eventType)
+			return false
+		}
+
+		clusterName, err := names.GetClusterName(obj.Namespace)
+		if err != nil {
+			klog.Errorf("Failed to get member cluster name for work %s/%s", obj.Namespace, obj.Name)
+			return false
+		}
 
-			clusterName, err := names.GetClusterName(obj.Namespace)
-			if err != nil {
-				klog.Errorf("Failed to get member cluster name for work %s/%s", obj.Namespace, obj.Name)
-				return false
-			}
+		clusterObj, err := util.GetCluster(mgr.GetClient(), clusterName)
+		if err != nil {
+			klog.Errorf("Failed to get the given member cluster %s", clusterName)
+			return false
+		}
 
-			clusterObj, err := util.GetCluster(mgr.GetClient(), clusterName)
-			if err != nil {
-				klog.Errorf("Failed to get the given member cluster %s", clusterName)
-				return false
-			}
+		return clusterObj.Spec.SyncMode == clusterv1alpha1.Push
+	}
 
-			return clusterObj.Spec.SyncMode == clusterv1alpha1.Push
+	return predicate.Funcs{
+		CreateFunc: func(createEvent event.CreateEvent) bool {
+			return predFunc("create", createEvent.Object)
 		},
 		UpdateFunc: func(updateEvent event.UpdateEvent) bool {
-			obj := updateEvent.ObjectNew.(*workv1alpha1.Work)
-
-			// Ignore the object that has been suppressed.
-			if util.GetLabelValue(obj.Labels, util.PropagationInstruction) == util.PropagationInstructionSuppressed {
-				klog.V(5).Infof("Ignored Work(%s/%s) update event as propagation instruction is suppressed.", obj.Namespace, obj.Name)
-				return false
-			}
-
-			clusterName, err := names.GetClusterName(obj.Namespace)
-			if err != nil {
-				klog.Errorf("Failed to get member cluster name for work %s/%s", obj.Namespace, obj.Name)
-				return false
-			}
-
-			clusterObj, err := util.GetCluster(mgr.GetClient(), clusterName)
-			if err != nil {
-				klog.Errorf("Failed to get the given member cluster %s", clusterName)
-				return false
-			}
-
-			return clusterObj.Spec.SyncMode == clusterv1alpha1.Push
+			return predFunc("update", updateEvent.ObjectNew) || predFunc("update", updateEvent.ObjectOld)
```

Comment:
I think the original logic is easier to understand.

Reply:
The code in `UpdateFunc` and `DeleteFunc` are same, so I extract it.